R/tm_a_regression.R

```diff
@@ -1037,25 +1033,52 @@ srv_a_regression <- function(id,
     )
 
     ### REPORTER
-    if (with_reporter) {
-      card_fun <- function(comment, label) {
-        card <- teal::report_card_template(
-          title = "Linear Regression Plot",
-          label = label,
-          with_filter = with_filter,
-          filter_panel_api = filter_panel_api
-        )
-        card$append_text("Plot", "header3")
-        card$append_plot(plot_r(), dim = pws$dim())
-        if (!comment == "") {
-          card$append_text("Comment", "header3")
-          card$append_text(comment)
-        }
-        card$append_src(source_code_r())
-        card
-      }
-      teal.reporter::simple_reporter_srv("simple_reporter", reporter = reporter, card_fun = card_fun)
-    }
+    card_fun <- reactive({
+      req(plot_r(), source_code_r())
+      teal.reporter::report_document(
+
+        "## Plot",
+        plot_to_base64(plot_r()),
+        "## Source Code",
+        format_as_markdown_chunk(source_code_r()),
+        "## Table for testing",
+        knitr::kable(head(iris))
+      )
+    })
     ###
+
+    list(
+      report_card = card_fun
+    )
   })
 }
+
+
+plot_to_base64 <- function(plot, width = 5, height = 4, dpi = 100) {
+  # Temporary file to save the plot
+  tmpfile <- tempfile(fileext = ".png")
+
+  # Save the plot as a PNG file
+  ggsave(tmpfile, plot = plot, width = width, height = height, dpi = dpi)
+
+  # Read the binary data and encode as base64
+  # base64enc::base64encode(tmpfile)
+  base64_string<- knitr::image_uri(tmpfile)
+  sprintf("![Plot](%s)", base64_string)
+}
+#
+# plot_to_base64(qqplot(x = 1:2, y = 1:2))
+
+format_as_markdown_chunk <- function(r_code) {
+  # Wrap the code inside a Markdown R chunk
+  sprintf("```{r}\n%s\n```", r_code)
```

Comment:
```suggestion
  sprintf("```{r, eval=FALSE}\n%s\n```", r_code)
```

Reply:
I think eval=FALSE or any other chunk parameter shouldn't be given by default. Parameters should be arguments of the `as.code_block` function. I'm not sure at this point whether "code_block_elem" should be kept in the list as-is with the special class or if the backticks-enclosure should be added immediately.